src/physics/model.cpp

```diff
@@ -15,11 +15,92 @@ template <int dim, int nstate, typename real>
 ModelBase<dim, nstate, real>::ModelBase(
     std::shared_ptr< ManufacturedSolutionFunction<dim,real> > manufactured_solution_function_input):
         manufactured_solution_function(manufactured_solution_function_input)
-{ }
+{ 
+    // if provided with a null ptr, give it the default manufactured solution
+    // currently only necessary for the unit test
+    if(!manufactured_solution_function)
+        manufactured_solution_function = std::make_shared<ManufacturedSolutionSine<dim,real>>(nstate);
+}
 //----------------------------------------------------------------
 template <int dim, int nstate, typename real>
 ModelBase<dim,nstate,real>::~ModelBase() {}
 //----------------------------------------------------------------
+template <int dim, int nstate, typename real>
+void ModelBase<dim, nstate, real>
+::boundary_face_values (
+   const int /*boundary_type*/,
+   const dealii::Point<dim, real> &pos,
+   const dealii::Tensor<1,dim,real> &normal_int,
+   const std::array<real,nstate> &soln_int,
+   const std::array<dealii::Tensor<1,dim,real>,nstate> &soln_grad_int,
+   std::array<real,nstate> &soln_bc,
+   std::array<dealii::Tensor<1,dim,real>,nstate> &soln_grad_bc) const
+{
+    std::array<real,nstate> boundary_values;
+    std::array<dealii::Tensor<1,dim,real>,nstate> boundary_gradients;
+    for (int s=0; s<nstate; s++) {
+        boundary_values[s] = this->manufactured_solution_function->value (pos, s);
+        boundary_gradients[s] = this->manufactured_solution_function->gradient (pos, s);
+    }
+
+    for (int istate=0; istate<dim+2; ++istate) {
+        soln_bc[istate] = 0.0;
+        soln_grad_bc[istate] = 0.0;
+    }
+    for (int istate=dim+2; istate<nstate; ++istate) {
+
+        std::array<real,nstate> characteristic_dot_n = convective_eigenvalues(boundary_values, normal_int);
+        const bool inflow = (characteristic_dot_n[istate] <= 0.);
+
+        if (inflow) { // Dirichlet boundary condition
+            soln_bc[istate] = boundary_values[istate];
+            soln_grad_bc[istate] = soln_grad_int[istate];
+        } else { // Neumann boundary condition
+            soln_bc[istate] = soln_int[istate];
+            soln_grad_bc[istate] = soln_grad_int[istate];
+        }
+    }
```

Comment:
one problem with this is that it doesn't include all the boundary IDs/conditions like in `euler.cpp`; so you'll get errors if you try to run anything other than a manufactured_solution boundary condition

Reply:
fixed.

Reply:
Now the boundary_face_value is still defined in the model.cpp so that LES would not be affected.